heft-plugins/heft-lint-plugin/src/Eslint.ts

```diff
@@ -166,17 +176,10 @@ export class Eslint extends LinterBase<TEslint.ESLint.LintResult> {
           return lintResult.fixableErrorCount + lintResult.fixableWarningCount > 0;
         }));
 
-    const failures: TEslint.ESLint.LintResult[] = [];
-    for (const lintResult of lintResults) {
-      if (lintResult.messages.length > 0 || lintResult.output) {
-        failures.push(lintResult);
-      }
-    }
-
-    return failures;
+    return lintResults;
```

Comment:
This seems like a big change in behavior. What is the net result of this? Are we breaking anything downstream of this method by including all entries now instead of only the ones with an error message/output?

Reply:
This change won't break anything downstream. Where this function is called, there is an additional filter that parses by lintMessage so if an entry doesn't contain an error message/output, that entry is skipped so nothing gets broken downstream.

Reply:
In this case, we should update the variable names here to ensure that it's clear that this is all results, not just failures: https://github.com/microsoft/rushstack/blob/982df68fdbfb6711a9371bf6662d313419aaf9ad/heft-plugins/heft-lint-plugin/src/LinterBase.ts#L150

Reply:
The one thing this does affect is the incremental cache file, which will now always map to a (possibly empty) list of existing messages, but the observed behavior to the user should be equivalent.